src/Environment.jl

```diff
@@ -0,0 +1,42 @@
+
+const _NREFS = Ref(0)
+const _INITIALIZED = Ref(false)
+
+function Init(;p4est_verbosity_level=P4est_wrapper.SC_LP_DEFAULT,finalize_atexit=true)
+  if !MPI.Initialized()
+      MPI.Init()
+  end
+  if finalize_atexit
+    atexit(Finalize)
+  end
+  Finalize()
+
+  sc_init(parts.comm, Cint(true), Cint(true), C_NULL, p4est_verbosity_level)
+  p4est_init(C_NULL, p4est_verbosity_level)
+  _INITIALIZED[] = true
+
+  return nothing
+end
+
+function Initialized()
+  return _INITIALIZED[]
+end
+
+function Finalize()
+  if Initialized()
+    GC.gc() # Finalize all object out of scope at this point
+    if _NREFS[] != 0
+      @warn "$(_NREFS[]) objects still not finalized before calling GridapP4est.Finalize()"
+    end
+    _NREFS[] = 0
+    sc_finalize()
+  end
+  return nothing
+end
+
+function with(f;kwargs...)
```

Comment:
are we currently calling this function in any of the tests? 

I think that GridapPETSc.jl has a similar function. How these two compose together in an scenario in which you use both packages in the same driver?

Reply:
I think GridapPETSc does not export `with`, so that you have to explicitly call `GridapPETSc.with(...)`. I was planning on doing the same for GridapP4est. Then I believe both functions will be considered as completely different by the compiler.

Reply:
I was not actually referring to the name clash. (which of course is an issue as well which can be tackled as you say). 

I was thinking in how the two `with` functions compose with each other. Can be they called in a nested fashion? Does the nesting order matters? All these things I am wondering.